src/Symfony/Component/Security/Core/Authorization/Voter/AbstractVoter.php

```diff
@@ -83,18 +88,76 @@ public function vote(TokenInterface $token, $object, array $attributes)
     }
 
     /**
-     * Return an array of supported classes. This will be called by supportsClass
+     * Determines if the attribute and class are supported by this voter.
+     *
+     * To determine if the passed class is instance of the supported class, the
+     * isClassInstanceOf() method can be used.
+     *
+     * @param string $attribute An attribute
+     * @param string $class     The fully qualified class name of the passed object
+     *
+     * @return bool True if the attribute and class is supported, false otherwise
+     */
+    protected function supports($attribute, $class)
+    {
+        @trigger_error('The getSupportedClasses and getSupportedAttributes methods are deprecated since version 2.8 and will be removed in version 3.0. Overwrite supports instead.');
```

Comment:
People have to override the supports method to see if the attribute and class are supported. If they didn't, this code will be executed and they will recieve a deprecation notice. If they did, this code is overriden and the deprecation notice isn't triggered.

The triggers in the `getSupported*` classes are there in case someone is executing this methods directly, to indicate that they have to use `supports`.